src/scala/codeGeneration.js

```diff
@@ -60,7 +60,7 @@ export function generateSource(context, options) {
   generator.printOnNewline('//  This file was automatically generated and should not be edited.');
   generator.printNewline();
 
-  if (context.namespace) {
+  if (context.options.namespace) {
```

Comment:
Would a test be good for this? Causes a headache when it breaks! Happy to commit to this PR and see what you think?

Reply:
Got some free time today, so I went ahead and converted everything to TypeScript and added a test for adding the package declaration!

Reply:
Nice!